en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md

```diff
@@ -0,0 +1,341 @@
+# Write a Custom OAuth 2.0 Grant Type
+
+OAuth 2.0 authorization servers provide support for four main grant
+types according to the [OAuth 2.0
+specification](https://tools.ietf.org/html/rfc6749). It
+also has the flexibility to support any custom grant types. This topic
+provides instructions on how to implement a custom grant type for OAuth
+2.0 authorization server and how to extend the behavior of default grant
+types.
+
+The WSO2 Identity Server is used as the OAuth 2.0 authorization server
+implementation, which is an open source implementation.
+
+------------------------------------------------------------------------
+
+## Implement a new grant type
+
+Follow the steps given below to implement a new grant type.
+
+1.  Implement the following two extensions.
+    -   `            GrantTypeHandler           ` - This is the
+        implementation of the grant type. Here you can implement the
+        way, it must be validated and how token must be issued. You can
+        write the new implementation by implementing the 
+        `            AuthorizationGrantHandler           `  interface
+        or by extending 
+        `            AbstractAuthorizationGrantHandler           ` . In
+        most cases, it is enough to extend the 
+        `            AbstractAuthorizationGrantHandler           `  in
+        the WSO2 OAuth component.
+    -   `            GrantTypeValidator           ` - This is used to
+        validate the grant request that is sent to the
+        `            /token           ` endpoint. You can define what
+        parameters must be in the request and define the validation of
+        them. You can write the new implementation by extending the 
+        `            AbstractValidator           `  in Apache Amber
+        component.
+2.  When implementation is done, package your class as a .jar file and
+    place it in the
+    `          <IS_HOME>/repository/component/lib         ` directory.
```

Comment:
shouldn't it need to be put into dropins?

Reply:
This doesn't work when the .jar file is put into dropins.